### PR TITLE
LOG-2587: Fix ES functional tests for vector, using go tags

### PR DESCRIPTION
**Pull request by vimalk78**
### Description
This PR

  - fixes functional framework to make `kubernetes_logs` start scraping application logs created by functional framework, enables debug logs for vector
  - adds go test tags based selection of collector for functional test framework.
    - use `go test --tag=fluentd` to set a global variable `"FrameworkType"` to "fluentd"
    - use `go test --tag=vector` to set a global variable `"FrameworkType"` to "vector"
    which can be used to create framework in the tests.
  - fixes app log data model to conform to functional test app log template
  - relaxes message template to accommodate both fluentd and vector based record creation
  - enables outputs/elasticsearch tests to run for both fluentd and vector

/cc @jcantrill @cahartma 

/assign @vimalk78 

### Links
- JIRA: https://issues.redhat.com/browse/LOG-2587

### CODE REVIEW

**Review by jcantrill — CHANGES_REQUESTED** [review]
/hold

I think this is a good direction that would actually be better solved with gingkgo 2.0 and labels, but that being said I think it's still missing some meta to properly use tags.  The way I read the use of tags is that we would wish to annotate the test files with the desired tag so that it is compiled and runs the way we require.  I only see tags in the marker files for which collector type; that does not look to be enough. 

Another point is why do we need "if" statements to skip if we have the tags;  we should be able to "skip" those tests by marking them ready for fluentd and vector

File: internal/runtime/pod.go
Comment (vimalk78) [reply internal/runtime/pod.go:103]: i observed test speedup once i changed this to `IfNotPresent` 
since tests run in parallel, multiple pods start pulling the same image, and it seemed to slow down the tests.

i also used a "warm up" pod which has containers for each image tests needed. 
```
  containers:    
    - name: c0    
      image: elasticsearch:7.10.1    
    - name: c1    
      image: quay.io/openshift/origin-logging-rsyslog:latest    
    - name: c2    
      image: quay.io/openshift-logging/fluentd:1.14.5    
etc
```
then tests didnt have to wait for pods to pull the images.

File: test/functional/fluentd.go
Comment (alanconway) [inline test/functional/fluentd.go:11]:
```suggestion
	LogCollectionType = logging.LogCollectionTypeFluentd
```

**Review by alanconway — CHANGES_REQUESTED** [review]
I like it, just change "FrameworkType" to "LogCollectionType" - framework is confusing in this context.

File: test/functional/fluentd.go
Comment (vimalk78) [reply test/functional/fluentd.go]: @alanconway Updated

File: internal/generator/vector/conf_test.go
Comment (cahartma) [inline internal/generator/vector/conf_test.go:162]: you should be able to shorten this to:  `."@timestamp = del(.timestamp) `, since del() returns the value of the deleted field.

**Review by alanconway — APPROVED** [review]
/lgtm

File: internal/generator/vector/conf_test.go
Comment (vimalk78) [reply internal/generator/vector/conf_test.go]: @cahartma updated PR

Comment (vimalk78) [reply internal/generator/vector/conf_test.go]: will need re-lgtm though

**Review by jcantrill — COMMENTED** [review]
/approve
/hold cancel

**Pull request merged**
